### PR TITLE
research 2019-19 add to calendar update research and prototype 4

### PR DESCRIPTION
**Pull request by Ricokola**

### CODE REVIEW

**Comment by Ricokola** [comment]
Update JS script and link to prototype 4

**Comment by Ricokola** [comment]
Update prototype 4

File: research/2019-19-add-calendar-prototype-4.md [hunk 1]
```diff
@@ -53,22 +53,48 @@ Prototype of an ```Add to Calendar``` with the details summary pattern and Schem
...
-				<h2 property="name">Canux20</h2>
-				<p property="description">CanUX (pronounced “canucks”, like the hockey team) is an independent, locally-grown, non-profit, volunteer-driven event created for people who want to learn and share their expertise about user experience, and how our profession contributes to making things work better for everyone.</p>
-				<h3>Date and time</h3>	
-				<p>Start date: <time property="startDate" content="20200127T120000Z" datetime="2020-01-27 7:00">Mon, 27 Jan 2020, 7:00 AM EST</time><br>
-				End date: <time property="endDate" content="20200127T170000Z" datetime="2020-01-29 12:00">Wed, 29 Jan 2020, 12:00 PM EST</time></p>
-				<h3>Location</h3>
-				<P><a href="https://www.historymuseum.ca/" class="wb-addcal-location">Canadian <span lang="en">Museum</a< of History</a></p>
-			</div>
-		</section>	
+<div class="wb-addcal mrgn-bttm-md" typeof="Event">
```
Comment (duboisp) [inline research/2019-19-add-calendar-prototype-4.md]: The css class provisional is missing
```suggestion
<div class="provisional wb-addcal mrgn-bttm-md" typeof="Event">
```

File: research/2019-assets/add-to-calendar/1-add-to-calendar.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+		// Start initialization
+		// returns DOM object = proceed with init
+		// returns undefined = do not proceed with init (e.g., already initialized)
+		var elm = wb.init( event, componentName, selector ),
+			$elm = $( elm );
+
+		if ( elm ) {
+			
+			var properties = elm.querySelectorAll( "[property]" ),
+				i,
```
Comment (duboisp) [inline research/2019-assets/add-to-calendar/1-add-to-calendar.js]:
```
prop_cache,
prop_val_cache,
```

File: research/2019-assets/add-to-calendar/1-add-to-calendar.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+			// Initiate dictionary
+			i18nDict = i18nDict[ $( "html" ).attr( "lang" ) || "en" ];
+			i18nDict = {
+				addto: i18nDict[ "addcal-addto" ],
+				calendar: i18nDict[ "addcal-calendar" ],
+				ical: i18nDict[ "addcal-ical" ]
+			};
+		
+			for ( i=0; i < i_len; i++ ) {
+				switch ( properties[ i ].getAttribute( "property" ) ) {
```
Comment (duboisp) [inline research/2019-assets/add-to-calendar/1-add-to-calendar.js]: ```prop_cache = properties[ i ];```

File: research/2019-assets/add-to-calendar/1-add-to-calendar.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+					case "description":
+						eventData.description = properties[ i ].innerText;
+						googleLink += "&details=" + eventData.description;
+						icsLink += "\nDESCRIPTION:" + eventData.description.replace( /(\r\n|\n|\r)/gm, " " );
+						break;
+					case "startDate":
+						eventData.startDate = properties[ i ].getAttribute( "content" );
+						googleLink += "&dates=" + eventData.startDate;
+						icsLink += "\nDTSTART:" + eventData.startDate;
+						break;
```
Comment (duboisp) [inline research/2019-assets/add-to-calendar/1-add-to-calendar.js]:
```
prop_val_cache = prop_cache.getAttribute( "content" );
googleLink += "&dates=" + prop_val_cache ;
icsLink += "\nDTSTART:" + prop_val_cache;
```

Current when minified:
```
case "startDate":
a.startDate = b[ i ].getAttribute( "content" );
c += "&dates=" + a.startDate;
d += "\nDTSTART:" + a.startDate;
break;
```

Optimized version:
```
case "startDate":
a = b.getAttribute( "content" );
c += "&dates=" + a ;
d += "\nDTSTART:" + a;
```

File: research/2019-assets/add-to-calendar/1-add-to-calendar.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+					case "endDate":
+						eventData.endDate = properties[ i ].getAttribute( "content" );
+						googleLink += "/" + eventData.endDate;
+						icsLink += "\nDTEND:" + eventData.endDate;
+						break;
+					case "streetAddress":
+						eventData.streetAddress = properties[ i ].innerText;
+						googleLink += " " + eventData.streetAddress;
+						icsLink += " " + eventData.streetAddress;
+						break;
```
Comment (duboisp) [inline research/2019-assets/add-to-calendar/1-add-to-calendar.js]:
```
case "streetAddress":
prop_val_cache = " " + prop_cache.innerText;
googleLink += prop_val_cache ;
icsLink += prop_val_cache;
```

File: research/2019-assets/add-to-calendar/1-add-to-calendar.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+		var elm = wb.init( event, componentName, selector ),
+			$elm = $( elm );
+
+		if ( elm ) {
+			
+			var properties = elm.querySelectorAll( "[property]" ),
+				i,
+				i_len = properties.length,
+				eventData = {},
+				place = false,
```
Comment (duboisp) [inline research/2019-assets/add-to-calendar/1-add-to-calendar.js]: Pas besoin d'être définie à false.
`Undefined === null === false`

```
place, // false
```

File: research/2019-assets/add-to-calendar/1-add-to-calendar.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+"use strict";
+
+/*
+ * Variable and function definitions.
+ * These are global to the plugin - meaning that they will be initialized once per page,
+ * not once per instance of plugin on the page. So, this is a good place to define
+ * variables that are common to all instances of the plugin on a page.
+ */
+var componentName = "wb-addcal",
+	selector = ".provisional." + componentName,
```
Comment (duboisp) [inline research/2019-assets/add-to-calendar/1-add-to-calendar.js]:
```suggestion
	selector = ".provisional[typeof=Event]." + componentName,
```

**Pull request merged**